x/interchainqueries/keeper/msg_server.go

```diff
@@ -234,14 +247,52 @@ func (k msgServer) SubmitQueryResult(goCtx context.Context, msg *types.MsgSubmit
 		if !types.InterchainQueryType(query.QueryType).IsTX() {
 			return nil, sdkerrors.Wrapf(types.ErrInvalidType, "invalid query result for query type: %s", query.QueryType)
 		}
+
 		if err := k.ProcessBlock(ctx, queryOwner, msg.QueryId, msg.ClientId, msg.Result.Block); err != nil {
 			ctx.Logger().Debug("SubmitQueryResult: failed to ProcessBlock",
 				"error", err, "query", query, "message", msg)
 			return nil, sdkerrors.Wrapf(err, "failed to ProcessBlock: %v", err)
 		}
+
+		if err = k.UpdateLastLocalHeight(ctx, query.Id, uint64(ctx.BlockHeight())); err != nil {
+			return nil, sdkerrors.Wrapf(err,
+				"failed to update last local height for a result with id %d: %v", query.Id, err)
+		}
 	}
 
 	return &types.MsgSubmitQueryResultResponse{}, nil
 }
 
+func getEventsQueryUpdated(query *types.RegisteredQuery) sdk.Events {
+	return sdk.Events{
+		sdk.NewEvent(
+			types.EventTypeNeutronMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+			sdk.NewAttribute(sdk.AttributeKeyAction, types.AttributeValueQueryUpdated),
+			sdk.NewAttribute(types.AttributeKeyQueryID, strconv.FormatUint(query.Id, 10)),
+			sdk.NewAttribute(types.AttributeKeyConnectionID, query.ConnectionId),
+			sdk.NewAttribute(types.AttributeKeyOwner, query.Owner),
+			sdk.NewAttribute(types.AttributeKeyQueryType, query.QueryType),
+			sdk.NewAttribute(types.AttributeTransactionsFilterQuery, query.TransactionsFilter),
+			sdk.NewAttribute(types.AttributeKeyKVQuery, types.KVKeys(query.Keys).String()),
+		),
+	}
+}
+
+func getEventsQueryRemoved(query *types.RegisteredQuery) sdk.Events {
+	return sdk.Events{
+		sdk.NewEvent(
+			types.EventTypeNeutronMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+			sdk.NewAttribute(sdk.AttributeKeyAction, types.AttributeValueQueryRemoved),
+			sdk.NewAttribute(types.AttributeKeyQueryID, strconv.FormatUint(query.Id, 10)),
+			sdk.NewAttribute(types.AttributeKeyConnectionID, query.ConnectionId),
```

Comment:
I dont know why do we need all these attributes for the removeQuery event, but if so, can we create a more generic function for both `getEventsQueryUpdated` and `getEventsQueryUpdated` almost all attributes the same

Reply:
removed most of the attributes from the remove action